openhands/agenthub/codeact_agent/codeact_agent.py

```diff
@@ -482,7 +482,14 @@ def _get_messages(self, state: State) -> list[Message]:
                         and messages[-1].role == message.role
                         and message.role != 'tool'
                     ):
-                        messages[-1].content.extend(message.content)
+                        # find the last TextContent object in the last message
+                        last_text_content = None
+                        for content in reversed(messages[-1].content):
+                            if isinstance(content, TextContent):
+                                last_text_content = content
+                                break
+                        if last_text_content:
```

Comment:
If we can't find the last_text_content, does it means that we will remove the new message?

Reply:
Fixing!

Reply:
By the way, we probably shouldn't be doing this kind of thing at all anymore... Anthropic finally doesn't need it anymore, it accepts u/u/a/a. And litellm is doing it for providers that still need it.

Reply:
Yeah! Let's remove it then - it makes our code much more ugly :(